native/cocos/engine/Engine.cpp

```diff
@@ -285,6 +301,13 @@ void Engine::tick() {
 
         prevTime = std::chrono::steady_clock::now();
 
+#if WATCHDOG_ENABLED
+        WatchDog watchdog(
```

Comment:
Please don't create it if not needed.

Reply:
The blockingTimeout object is created at runtime, and it cannot be controlled through macros. When it is disabled, the overhead of this object is already small.

Reply:
Otherwise, we need consider timeout to call new/delete. That can be more expensive.